haskell-style.md

```diff
@@ -519,132 +219,38 @@ main = do
+type, import the type un-qualified and the rest of the module qualified:
 
-```
+```hs
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
 import Data.Map (Map)
 import qualified Data.Map as Map
+import Data.Text (Text)
+import qualified Data.Text as T
 ```
 
 ### Abbreviating qualifications
 
-There are a number of common abbreviations that are used in the community to
-qualify imports.
-
-```haskell
-import qualified Data.Text as T
-import qualified Data.Text.Encoding as T
-
-import qualified Data.Text.Lazy as TL
-import qualified Data.Text.Lazy.Encoding as TL
-
-import qualified Data.ByteString as BS
-import qualified Data.ByteString.Lazy as BSL
-import qualified Data.ByteString.Char8 as BS8
-import qualified Data.ByteString.Lazy.Char8 as BSL8
-
-import qualified Data.List.NonEmpty as NE
-
-import qualified Data.Sequence as Seq
-```
+**Automated**.
 
 ### Import groups
 
 Put one blank line between `module`-`where` and the start of your `import`s. Put
 your preferred prelude (when explicit) first, followed by a blank line, then the
 rest of your imports.
 
-The main `import` group should be maintained by our `stylish-haskell`
-configuration.
-
-For vim users,
-
-```vim
-:stylish-haskell %
-
-" or visually select the imports and
-:'<,'>!stylish-haskell
-```
-
-An example of its results at the time of this writing is shown below, but what
-it actually does is less important than the fact that it's automated.
-
-```haskell
--- Bad
--- Improper spacing, improper sorting
-module Foo
-  ( bar
-  , baz
-  ) where
-import qualified Data.Map as Map
-import TextAssets.S3
-import Data.Text (Text)
-import Unit
-import Json
-import Control.Lens
-import qualified Data.Set as Set
-import ClassyPrelude
-import Network.AWS.S3
-import Data.Set (Set)
-import qualified Data.Text as T
-import Network.AWS
-import Data.Map (Map)
-import Data.Conduit
-
--- Good
-module Foo
-  ( bar
-  , baz
-  ) where
-
-import ClassyPrelude
-
-import Control.Lens
-import Data.Conduit
-import Data.Map (Map)
-import qualified Data.Map as Map
-import Data.Set (Set)
-import qualified Data.Set as Set
-import Data.Text (Text)
-import qualified Data.Text as T
-import Json (Json)
-import qualified Json as J
-import Network.AWS
-import Network.AWS.S3
-import TextAssets.S3
-import Unit
-```
-
 ## Exports
 
-Use sorted, multi-line exports. There are two exceptions to this rule:
-
-1. A single-line export for `Main`, when it only exports `main`:
-
-   ```haskell
-   module Main (main) where
-   ```
+**Automated**: the formatting of `module (...) where` itself.
 
-1. If the order of exports matters in your desired Haddock output, you may
-   violate sorting to achieve it.
-
-Otherwise:
-
-```haskell
--- Bad
-module Driver (scienceOptions, socialStudiesOptions, mainWith) where
-
--- Good
-module Driver
-  ( mainWith
-  , scienceOptions
-  , socialStudiesOptions
-  ) where
-```
+Sort your exports, unless the order matters in your desired Haddock output.
```

Comment:
The more I think about it, the more I think I'd remove this. In actuality, exports ordering should *always* be adjusted for the best Haddocks, even in an otherwise-undocumented module. I think our Haddock-related section could imply or out-right state to do so. :man_shrugging:

Reply:
Agreed, the order should not be arbitrary and I consider sorting arbitrary.

Reply:
So noted. I'll keep it for now and address in a separate PR.